### ISSUE TITLE
ADIOS2_USE_Endian_Reverse Broken

### ISSUE THREAD

**Issue by ax3l** [ornladios/ADIOS2#2700]
@eschnett reported in Spack that `-DADIOS2_USE_Endian_Reverse=ON` is broken: https://github.com/spack/spack/issues/23643
```
adios2/helper/adiosMemory.cpp:32:22: error: no match for 'operator==' (operand types are 'const adios2::DataType' and 'const char [1]')
   32 |         if (destType == "")
      |             ~~~~~~~~ ^~ ~~
      |             |           |
      |             |           const char [1]
      |             const adios2::DataType
```

I can reproduce this with the latest `master` as of d6547823360fe721ac065d731af4404ff07ff174

```
cmake -S . -B build -DADIOS2_USE_Endian_Reverse=ON
cmake --build build
```

cc @pnorbert @chuckatkins

**Comment by ax3l** [comment]
I guess  the right comparison would be something like `destType == DataType::None` or so?

**Comment by germasch** [comment]
Yes, this should be `destType == DataType::None`, and it might even be that it was me who broke it. If so, a real long time ago...

**Comment by ax3l** [comment]
Thanks, fix in #2701 :)
